src/snowflake/connector/connection.py

```diff
@@ -275,8 +278,17 @@ class SnowflakeConnection:
         schema: Schema in use on Snowflake.
         warehouse: Warehouse to be used on Snowflake.
         role: Role in use on Snowflake.
-        login_timeout: Login timeout in seconds. Used while authenticating.
-        network_timeout: Network timeout. Used for general purpose.
+        login_timeout: Login timeout in seconds. Login requests will not be retried after this timeout expires.
+            Note that the login attempt may still take more than login_timeout seconds as an ongoing login request
+            cannot be canceled even upon login timeout expiry. The login timeout only prevents further retries.
+        network_timeout: Network timeout in seconds. Network requests besides login requests will not be retried
+            after this timeout expires. Overriden in cursor query execution if timeout is passed to cursor.execute.
+            Note that an operation may still take more than network_timeout seconds for the same reason as above.
+        socket_timeout: Socket timeout in seconds. Sets both socket connect and read timeout.
```

Comment:
I'm wondering do we really need to introduce socket_timeout. it feels to me that `network_timeout` ~= `socket_timeout` conceptually

Reply:
I think there are some differences but I am not confident. Let's discuss this on Monday?

Reply:
Please post what you agreed on

Reply:
We retained `socket_timeout` but agreed that `network_timeout` and `socket_timeout` sounded too similar. Decided that `request_timeout` was more clear than `network_timeout`. Rebase overwrote my edit to the documentation string, but I've fixed that.